llumnix/global_scheduler/dispatch_policy.py

```diff
@@ -108,13 +123,66 @@ def dispatch(self,
         return target_instance_id
 
 
+class DynamicPD(DispatchPolicy):
+    def soft_dispatch(self,
+                      instance_type: InstanceType,
+                      prefill_instance_infos: Dict[str, InstanceInfo],
+                      prefill_instance_num_requests: Dict[str, int],
+                      decode_instance_infos: Dict[str, InstanceInfo],
+                      decode_instance_num_requests: Dict[str, int]) -> int:
+        if instance_type == InstanceType.PREFILL:
+            # if prefill loads are not busy, dispatch to prefill instance
+            prefill_sorted_instance_infos: List[InstanceInfo] = \
+                self.sort_instance_infos(prefill_instance_infos.values(), 'dispatch_load_metric')
+            if not prefill_sorted_instance_infos[0].dispatch_load_metric.is_busy():
```

Comment:
this is busy is not general enough. i think is_prefill_instances_busy should take all the prefill_instance_info as input, it will be more general.